widget/embedded/src/pages/SwapDetailsPage.tsx

```diff
@@ -1,19 +1,19 @@
 import { cancelSwap } from '@rango-dev/queue-manager-rango-preset';
 import { useManager } from '@rango-dev/queue-manager-react';
 import React from 'react';
+import { useParams } from 'react-router-dom';
 
 import { SwapDetails } from '../components/SwapDetails';
 import { SwapDetailsPlaceholder } from '../components/SwapDetails/SwapDetails.Placeholder';
 import { useNavigateBack } from '../hooks/useNavigateBack';
 import { useAppStore } from '../store/AppStore';
-import { useUiStore } from '../store/ui';
 import { getPendingSwaps } from '../utils/queue';
 
 export function SwapDetailsPage() {
   const { manager, state } = useManager();
   const loading = !state.loadedFromPersistor;
   const pendingSwaps = getPendingSwaps(manager);
-  const requestId = useUiStore.use.selectedSwapRequestId();
+  const { requestId } = useParams<{ requestId: string }>();
```

Comment:
I can't seem to find a situation where going to the "swap-details" page using React Router is possible with an empty string. In this case, the requestId || '' is just there to avoid TypeScript errors.